### ISSUE TITLE
Fix kube_events_total inconsistencies when isomorphic Events are created

### ISSUE THREAD

**Issue by dgrisonnet** [rhobs/kube-events-exporter#38]
### Description

When an Event is created, we increment the kube_events_total counter by 1. However, in some cases, a single Event creation can result in multiple isomorphic Events being created.

When an Event is updated, this is handled correctly because we add the difference between the counts of the old Event and the new one to the counter.

### Event creation

**"k8s.io/client-go/tools/record/" EventRecorder**

There are 2 cases where an Event is created with this recorder:
  - The Event count is equal to 1
  - The recorder tried to patch an Event that does not exist anymore on the apiserver

With this recorder, each Event generation results in one request to the apiserver. Thus, the assumption of one Create event resulting in one Event creation is correct.

**"k8s.io/client-go/tools/events/" EventRecorder**

There are 2 cases where an Event is created with this recorder:
  - The Event series is nil
  - The recorder tried to patch an Event that does not exist anymore on the apiserver

Both behaviour are really similar however, in this particular case, in order to reduce the number of call made to the apiserver, it was chosen to only update the Events on the apiserver every 30 minutes via a heartbeat (goroutine). Meaning that all updates during these 30 minutes gets merged into a single request. With a PATCH request, we won't have any problem to know how many Events were generated during this period since we will be able to get both the previous count and the new one. However, if the PATCH request fail, the Event will be recreated and we won't be able to know the previous count.
 
### Potential solutions

- When an Event is created add the Event count to the counter.

> We can't do that because, in the case of a deleted Event, the next update will result in this Event recreation with a count equal to all the older Events + the new ones.

> Example: Delete Ev{Count=100} (retention) => Update Ev{Count=101} => Fail (does not exist) => Create Ev{Count=101}

- When an Event is created for the first time if we increment the counter by one and store this Event + its count in an internal cache. We will then keep the data up to date in the internal cache and use it as the old Event count when updating the counter.

> We will have inconsistencies when multiple Events are created with the same apiserver request and seen for the first time by the exporter.

- Have a sync period before starting collecting metrics in order to save existing Event counts in an internal cache.

> This sync period needs to be longer than the period between 2 heartbeats (+30min). However, we won't have any inconsistencies with this method.

-- 
Current [event handler](https://github.com/rhobs/kube-events-exporter/blob/fdf768eafa4d96d541a1d5df516bb5497f88ed1e/internal/collector/event.go#L68-L84) in charge of updating the kube_events_total counter.

**Comment by dgrisonnet** [comment]
This data inconsistencies only happens during the exporter's first scrape and are not impacting metrics exposed by the exporter in the long run. The data loss is insignificant, so we should not introduce a complex mechanism to fix it. Thus, it is decent to keep it as is.
